src/shared/previewUtils.ts

```diff
@@ -58,38 +58,53 @@ export class PreviewUtils {
    * If it is in use then we increment the port number by 2 and check if it is in use or not.
    * This process is repeated until a port that is not in use is found.
    *
-   * @returns a port number to be used by the local dev server.
+   * @returns a pair of port numbers to be used by the local dev server for http and https.
    */
-  public static async getNextAvailablePort(): Promise<number> {
-    const userConfiguredPort = await ConfigUtils.getLocalDevServerPort();
+  public static async getNextAvailablePorts(): Promise<{ httpPort: number; httpsPort: number }> {
+    const userConfiguredPorts = await ConfigUtils.getLocalDevServerPorts();
 
-    if (userConfiguredPort) {
-      return Promise.resolve(userConfiguredPort);
+    if (userConfiguredPorts) {
+      return Promise.resolve(userConfiguredPorts);
     }
 
-    let port = LOCAL_DEV_SERVER_DEFAULT_PORT;
+    let httpPort = LOCAL_DEV_SERVER_DEFAULT_HTTP_PORT;
+    let httpsPort = httpPort + 1;
     let done = false;
 
     while (!done) {
-      const cmd =
-        process.platform === 'win32' ? `netstat -an | find "LISTENING" | find ":${port}"` : `lsof -i :${port}`;
+      const httpCmd =
+        process.platform === 'win32' ? `netstat -an | find "LISTENING" | find ":${httpPort}"` : `lsof -i :${httpPort}`;
+
+      const httpsCmd =
+        process.platform === 'win32'
+          ? `netstat -an | find "LISTENING" | find ":${httpsPort}"`
+          : `lsof -i :${httpsPort}`;
+
+      let httpResult = '';
+      let httpsResult = '';
 
       try {
-        const result = CommonUtils.executeCommandSync(cmd);
-        if (result.trim()) {
-          port = port + 2; // that port is in use so try another
-        } else {
-          done = true;
-        }
-      } catch (error) {
-        // On some platforms (like mac) if the command doesn't produce
-        // any results then that is considered an error but in our case
-        // that means the port is not in use and is ready for us to use.
+        httpResult = CommonUtils.executeCommandSync(httpCmd)?.trim();
+      } catch {
+        // ignore and continue
+      }
+
+      try {
+        httpsResult = CommonUtils.executeCommandSync(httpsCmd)?.trim();
+      } catch {
+        // ignore and continue
+      }
+
+      if (httpResult || httpsResult) {
```

Comment:
This logic is basically enforcing that these ports not only need to both be open, but sequential to each other. Wouldn't it make more sense to find the first open port, then find the next available one offset from there?

Reply:
Oh good point, let me do that